projects/output-and-error-handling/main.go

```diff
@@ -0,0 +1,96 @@
+package main
+
+import (
+	"bufio"
+	"fmt"
+	"net/http"
+	"os"
+	"time"
+)
+
+func sendRequest(url string) *http.Response {
+	resp, err := http.Get(url)
+	if err != nil {
+		fmt.Println("We can't get the weather!")
+		fmt.Fprintln(os.Stderr, "Failed to initialize.!", err)
+		os.Exit(2)
+	}
+	if resp == nil {
+		fmt.Fprintln(os.Stderr, "The server did not reply anything, which here is considered an error!")
+		os.Exit(52)
+	}
+	return resp
+}
+
+func handleStatusCode(resp *http.Response, sleepTime *time.Duration) {
+	defer resp.Body.Close()
+	if resp.StatusCode == 200 {
+		scanner := bufio.NewScanner(resp.Body)
+		for i := 0; scanner.Scan(); i++ {
+			fmt.Println(scanner.Text())
+		}
+		if err := scanner.Err(); err != nil {
+			panic(err)
+		}
+		os.Exit(0)
+	} else if resp.StatusCode == 429 {
+		// If I receive 429 but can't find ["Retry-After"] so I determine 1s to sleep. It act same as a while
+		retryAfter := resp.Header.Get("Retry-After")
+		if retryAfter == "" || retryAfter == "a while" {
```

Comment:
What if the header had "some time" or "asdbuibsafe" in it? What's special about these two values compared to any other value?

Reply:
I wanted to do early return just for these two cases. Others like "asdbuibsafe" will handle after failed in parsing.